### ISSUE TITLE
Display challenge results in race end card

### ISSUE THREAD

**Issue by Wuzzy2** [supertuxkart/stk-code#3877]
## Introduction
Currently (STK 1.0.0 and earliser), the interface isn't that very informative regarding the victory or failure of a Story Mode challenge. It can sometimes be pretty confusing as it's not clear when you failed.

## The problem
For instance, the end card does not tell the player directly if you failed or lost. You'd have to infer that from the buttons, but this is far from ideal. If you have completed a challenge *again*, the end card gives you no clue at all whether you have won or not, it looks the same. There's a lack of information.

Also, in some of the more difficult challenges it is not enough to finish first, you also must beat the clock. This can be easily missed, forgotten or misunderstood as it might be confusing that you don't unlock anything although you won the race. Improving the information would help here.

## Goal of this issue
To make it clearer to the player what has actually happened at the end of a challenge, whether you failed or not, and why you failed or won.

## Suggestions

I make the following suggestions for interface improvements:

1) In the race end card, show a line of text / title that says *directly* if you won or lost the challenge
2) In the race end card, show a list of all challenge objectives and which ones have been completed
3) 1) and 2) must work also after giving up a race. In this case, all challenge goals are counted as failures, no matter the actual progress

I have made a simple mock-up:
https://satoshiupload.com/images/2do3HV8eXb.png

In this mock-up, this would be the race end card that shows at the end of a nitro challenge which has failed. The icons and text color show the success or failure of each challenge objective. Also note the race records are gone to make space for the challenge results. Race records are not very important in this context and can be looked up elsewhere.

For the Grand Prix challenges, things are a bit different: Obviously, it doesn't make sense to show the challenge results after every race. Only after the final race, it makes sense to show the challenge results. Maybe in a separate screen after the end card of the final race.
However, I don't think this is important for now as all GP challenges are the same (just be 1st) and have no special requirement.
Long story short, it's OK to just leave the GP end carts unchanged for now.
But this is something to keep in mind for later. GP challenges might change in future should there ever be a GP challenge with a special goal (like a total time limit or a score goal).

**Comment by CodingJellyfish** [comment]
Close this issue please

**Comment by Benau** [comment]
maosen libera.chat banned in china? ![](https://cdn.lihkg.com/assets/faces/pig/arrested.gif)
